flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClient.java

```diff
@@ -275,9 +275,35 @@ public static ClassLoader retrieveClassLoader(
 	public static JobExecutionResult awaitJobResult(JobListeningContext listeningContext) throws JobExecutionException {
 
 		final JobID jobID = listeningContext.jobID;
+		final ActorRef jobClientActor = listeningContext.jobClientActor;
 		final Future<Object> jobSubmissionFuture = listeningContext.jobResultFuture;
 		final ClassLoader classLoader = listeningContext.classLoader;
 
+		while (!jobSubmissionFuture.isCompleted()) {
+			try {
+				Thread.sleep(250);
+			} catch (InterruptedException e) {
+				throw new JobExecutionException(jobID, "Interrupted while waiting for execution result.", e);
+			}
+
+			try {
+				Await.result(
+					Patterns.ask(
+						jobClientActor,
+						JobClientMessages.getPing(),
```

Comment:
I think we can also use Akka's build-in message `Identify` to do the same. Then we don't have to introduce a new message type.